### PR TITLE
Fix cookie SameSite warnings in Firefox console and hardcoded cookie name

### PR DESCRIPTION
**Pull request by dregad**
Fixes
- [#29611](https://www.mantisbt.org/bugs/view.php?id=29611) - samesite
- [#29616](https://www.mantisbt.org/bugs/view.php?id=29616) - hardcoded cookie name

### CODE REVIEW

File: config_defaults_inc.php [hunk 1]
```diff
@@ -3294,6 +3294,16 @@
  */
 $g_cookie_domain = '';
 
+/**
+ * Specifies the SameSite attribute to use for the MantisBT cookies.
+ *
+ * Valid values are Strict (default), Lax or None.
+ * @link https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+ *
+ * @global string $g_cookie_domain
```
Comment (atrol) [inline config_defaults_inc.php:3303]: Should be `$g_cookie_samesite`

File: config_defaults_inc.php [hunk 2]
```diff
@@ -4435,9 +4452,11 @@
 	'bug_list_cookie',
 	'cdn_enabled',
 	'class_path',
+	'collapse_settings_cookie',
 	'compress_html',
 	'cookie_domain',
 	'cookie_path',
+	'cookie_samesite',
```
Comment (atrol) [inline config_defaults_inc.php:4459]: Move after `cookie_prefix`

File: core/gpc_api.php [hunk 3]
```diff
@@ -384,31 +389,77 @@ function gpc_set_cookie( $p_name, $p_value, $p_expire = false, $p_path = null, $
 		$p_domain = config_get_global( 'cookie_domain' );
 	}
 
-	return setcookie( $p_name, $p_value, $p_expire, $p_path, $p_domain, $g_cookie_secure_flag_enabled, $p_httponly );
+	if( null === $p_samesite ) {
+		$p_samesite = config_get_global( 'cookie_samesite' );
+	}
+
+	if (PHP_VERSION_ID < 70300) {
```
Comment (atrol) [inline core/gpc_api.php:396]: Add blanks
`( PHP_VERSION_ID < 70300 )`

File: core/gpc_api.php [hunk 4]
```diff
@@ -384,31 +389,77 @@ function gpc_set_cookie( $p_name, $p_value, $p_expire = false, $p_path = null, $
...
 	}
 
 	# don't try to send cookie if headers are send (guideweb)
 	if( !headers_sent() ) {
-		return setcookie( $p_name, '', -1, $p_path, $p_domain );
+		# Note: This setcookie() call triggers a console warning in Firefox:
+		# Cookie “MANTIS_collapse_settings” has been rejected because it is already expired.
+		# apparently this is due to bug https://bugzilla.mozilla.org/show_bug.cgi?id=1676651
+
+		if (PHP_VERSION_ID < 70300) {
```
Comment (atrol) [inline core/gpc_api.php:446]: Add blanks
`( PHP_VERSION_ID < 70300 )`

File: core/gpc_api.php [hunk 5]
```diff
@@ -384,31 +389,77 @@ function gpc_set_cookie( $p_name, $p_value, $p_expire = false, $p_path = null, $
...
 
 	if( isset( $_COOKIE[$p_name] ) ) {
 		unset( $_COOKIE[$p_name] );
 	}
 
 	# don't try to send cookie if headers are send (guideweb)
 	if( !headers_sent() ) {
-		return setcookie( $p_name, '', -1, $p_path, $p_domain );
+		# Note: This setcookie() call triggers a console warning in Firefox:
+		# Cookie “MANTIS_collapse_settings” has been rejected because it is already expired.
```
Comment (atrol) [inline core/gpc_api.php:443]: Still using the former hardcoded name

**Review by atrol — APPROVED** [review]
Did some cursory tests using PHP 7.2 without getting any bugs.

File: core/gpc_api.php (hunk 5 above)
Comment (dregad) [reply core/gpc_api.php]: This is just a comment reflecting the cookie's default name as it appears in the Firefox console... Or am I missing something ?

Comment (dregad) [reply core/gpc_api.php]: I replaced `MANTIS` by `<PREFIX>` in the comment, assuming that's what you meant. If not, let me know.

**Pull request merged**